docs/source/_static/zzz_GENERATED_NOTEBOOK_SOURCE/tertiary/gwas.py

```diff
@@ -281,18 +276,16 @@ class Covariates:
 sample_info = get_sample_info(vcf_view, sample_metadata)
 sample_count = sample_info.count()
 mlflow.log_param("number of samples", sample_count)
-pcs_indexed = pcs_df.coalesce(1).withColumn("index", monotonically_increasing_id())
+pcs_indexed = pcs_df.coalesce(1).withColumn("index", fx.monotonically_increasing_id())
```

Comment:
It might make more sense to put the PCs into a CSV and just read that with pandas. I think that'll be a more common input format anyway.

Reply:
spark.read.format("delta").load(principal_components_path)
covariate_df = pd.DataFrame(covariates.head().covariates.toArray())

Reply:
This actually does not touch the datasource, and is only used for visualization. I'll switch to CSV in between though.